diff --git a/bsp-testkit/client/TestClient.scala b/bsp-testkit/client/TestClient.scala
index 404e349..f132ec1 100644
--- a/bsp-testkit/client/TestClient.scala
+++ b/bsp-testkit/client/TestClient.scala
@@ -7,9 +7,10 @@ import org.jetbrains.bsp.testkit.client.TestClient.{withLifetime, withSession}
 import org.jetbrains.bsp.testkit.JsonComparator
 
 import java.lang.reflect.Type
-import java.nio.file.Path
+import java.nio.file.{Path, Paths}
 import java.time.Duration
 import java.util.concurrent.ExecutionException
+import scala.concurrent.duration.DurationInt
 import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.jdk.CollectionConverters.{CollectionHasAsScala, SeqHasAsJava}
 import scala.jdk.DurationConverters._
@@ -65,7 +66,7 @@ class TestClient(val workspacePath: Path, val initializeParams: InitializeBuildP
     test: Session => Future[Unit]
   ): Unit = {
     implicit val ec: ExecutionContext = ExecutionContext.global
-    withSession(workspacePath, timeout, ignoreEarlyExit) { session =>
+    withSession(Paths.get(initializeParams.getRootUri), timeout, ignoreEarlyExit) { session =>
       withLifetime(initializeParams, session) { _ =>
         test(session)
       }
